src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,170 @@
+        logo = logo + "WHY ARE YOU BACK AGAIN ?_?";
+        description = logo;
+    }
+
+    /**
+     * Show line to the user.
+     */
+    public void showLine() {
+        description = "_______";
+    }
+
+    /**
+     * Show loading error to the user.
+     */
+    public void showLoadingError() {
+        description = "File not found.";
+    }
+
+
+    /**
+     * Show a message that a specified task has been deleted from the list.
+     * @param t The specified task deleted.
+     * @param taskList The task list the specified task is deleted from.
+     */
+    public void showDelete(Task t, TaskList taskList) {
+        String s = "";
+        s = "Noted. I've removed this task:\n";
+        s = s + t.toString() + "\n";
+        if (taskList.getTasksNumber() == 1) {
+            s = s + "Now you have 1 task in the list.\n";
+        } else {
+            s = s + "Now you have " + taskList.getTasksNumber() + " tasks in the list.\n";
+        }
+        description = s;
+
+    }
+
+    /**
+     * Show a message that a specified task has been added to the list.
+     * @param t The specified task to be added.
+     * @param taskList The task list the specified task is added to.
+     */
+    public void showAddTask(Task t, TaskList taskList) {
+        String s = "";
+        s = "Got it. I've added this task:\n";
+        s = s + t.toString() + "\n";
+        if (taskList.getTasksNumber() == 1) {
+            s = s + "Now you have 1 task in the list.\n";
+        } else {
+            s = s + "Now you have " + taskList.getTasksNumber() + " tasks in the list.\n";
+        }
+        description = s;
+    }
+
+    /**
+     * Show a message that a specified task has been marked done.
+     * @param t The specified task to be marked done.
+     */
+    public void showMark(Task t) {
+        String s = "Nice! I've marked this task as done:\n";
+        s = s + t + "\n";
+        description = s;
+    }
+
+    /**
+     * Show a message that a specified task has been marked undone.
+     * @param t The specified task to be marked undone.
+     */
+    public void showUnmark(Task t) {
+        String s = "Ok, I've marked this task as not done yet:\n";
+        s = s + t + "\n";
+        description = s;
+    }
+
+    /**
+     * Show all the tasks in the task list.
+     * @param taskList The specified task list.
+     */
+    public void showTasks(ArrayList<Task> taskList) {
+        String s = "";
+        s = "Here are the tasks in your list:\n";
+        for (int i = 0; i < taskList.size(); i++) {
+            Task currTask = taskList.get(i);
+            s = s + (i + 1) + "." + currTask.toString() + "\n";
+        }
+        description = s;
+    }
+
+    /**
+     * Show all tasks on a specified date.
+     * @param onDateTasks ArrayList that represents all tasks on date.
+     */
+    public void showGetDate(ArrayList<Task> onDateTasks) {
+        String s = "";
+        for (Task t : onDateTasks) {
+            s = s + t + "\n";
```

Comment:
Not a coding standard violation, but you can consider using StringBuilder to concatenate Strings!